src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSMove.java

```diff
@@ -57,8 +62,35 @@ public void run(ActionContext context) throws IOException {
     }
     StorageClient storageClient = StorageClient.create(config.getProject(), config.getServiceAccount(),
                                                        isServiceAccountFilePath);
-    //noinspection ConstantConditions
-    storageClient.move(config.getSourcePath(), config.getDestPath(), config.recursive, config.shouldOverwrite());
+
+    GCSPath destPath = config.getDestPath();
+    GCSPath undo = null;
+    Storage storage = storageClient.storage;
+
+    // create the destination bucket if not exist
+    Bucket bucket = null;
+    try {
+      bucket = storage.get(destPath.getBucket());
+    } catch (StorageException e) {
+      throw new RuntimeException(
+        String.format("Unable to access or create bucket %s. ", destPath.getBucket())
```

Comment:
similar comment about the message and similar question about whether this change means the plugin needs extra permissions